### PR TITLE
Enforce pedantic lints

### PR DESCRIPTION
**Pull request by natebosch**
Fix violations of:
- annotate_overrides
- prefer_single_quotes
- unawaited_futures

Disable implicit casts and enable strict raw types. These were already
clean.

Drop a small `unawaited` utility in this package rather than use the one
in `package:pedantic` to keep transitive deps minimal.

### CODE REVIEW

File: lib/src/utils.dart [hunk 1]
```diff
@@ -20,9 +20,11 @@ final Future<bool> supportsIPv6 = () async {
...
     var socket = await ServerSocket.bind(InternetAddress.loopbackIPv4, 0);
-    socket.close();
+    unawaited(socket.close());
     return true;
   } on SocketException catch (_) {
     return false;
   }
 }();
+
+void unawaited(Future<void> f) {}
```
Comment (kevmoo) [inline lib/src/utils.dart:31]: worth a `// NOTE` explaining why you're not using the impl in pedantic? (avoid more dependencies, I'm guessing...)

**Review by kevmoo — APPROVED** [review]

File: analysis_options.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+include: package:pedantic/analysis_options.yaml
+analyzer:
+  strong-mode:
+    implicit-casts: false
+  language:
+    strict-raw-types: true
```
Comment (kevmoo) [inline analysis_options.yaml:6]: Is this a "thing" we're doing now?

Comment (natebosch) [reply analysis_options.yaml:6]: 🤷‍♂ When it is easy it seems like a good idea.

`implicit-casts: false` will be at least partially enforced along with null safety. The difference is this way will also check casts from `dynamic` which upcoming Dart will continue to ignore, and checking that is still worth doing.

The `strict-raw-types` I'm less sure about the utility. I _think_ there will be a general trend in this direction, but I'd also be fine skipping it for now.

File: lib/src/utils.dart (hunk 1 above)
Comment (natebosch) [reply lib/src/utils.dart:31]: It's in the commit message and should show up with a `git blame` easy enough.

Comment (kevmoo) [reply lib/src/utils.dart:31]: 😕 – bleh. Don't make me use `git blame` when I'm scrolling through source!

Comment (natebosch) [reply lib/src/utils.dart:31]: IMO `git blame` is exactly the tool we should use to answer that type of question. It's a detail that is almost never relevant and does not alleviate the risk of introducing a bug.

**Comment by kevmoo** [comment]
hrm...

File: lib/src/utils.dart (hunk 1 above)
Comment (natebosch) [reply lib/src/utils.dart:31]: Added a comment.

**Pull request merged**